torch/onnx/ops/_impl.py

```diff
@@ -0,0 +1,317 @@
+                    encoded.attr_types.append("ts")
+                else:
+                    raise ValueError(f"Unsupported sequence type for attribute {k}")
+            else:
+                raise ValueError(f"Unsupported attribute type for {k}: {type(v)}")
+        assert len(encoded.attr_keys) == len(encoded.attr_types), (
+            f"Mismatch between number of attribute keys and types: {len(encoded.attr_keys)} != {len(encoded.attr_types)}"
+        )
+        assert len(encoded.attr_keys) == len(encoded.attr_pos), (
+            f"Mismatch between number of attribute keys and positions: {len(encoded.attr_keys)} != {len(encoded.attr_pos)}"
+        )
+        return encoded
+
+    def to_dict(self) -> dict[str, Union[int, float, str]]:
+        attrs = {}
+        for i, key in enumerate(self.attr_keys):
+            attr_type = self.attr_types[i]
+            if attr_type == "i":
+                attrs[key] = self.attr_ints[self.attr_pos[i][0]]
+            elif attr_type == "f":
+                attrs[key] = self.attr_floats[self.attr_pos[i][0]]
+            elif attr_type == "s":
+                attrs[key] = self.attr_strs[self.attr_pos[i][0]]
+            elif attr_type == "t":
+                attrs[key] = self.attr_tensors[self.attr_pos[i][0]]
+            elif attr_type == "fs":
+                attrs[key] = self.attr_floats[self.attr_pos[i][0] : self.attr_pos[i][1]]
+            elif attr_type == "is":
+                attrs[key] = self.attr_ints[self.attr_pos[i][0] : self.attr_pos[i][1]]
+            elif attr_type == "ss":
+                attrs[key] = self.attr_strs[self.attr_pos[i][0] : self.attr_pos[i][1]]
+            elif attr_type == "ts":
+                attrs[key] = self.attr_tensors[
+                    self.attr_pos[i][0] : self.attr_pos[i][1]
+                ]
+            else:
+                raise ValueError(f"Unsupported attribute type: {attr_type}")
+        return attrs
+
+
+@torch.library.custom_op(
+    "onnx_symbolic::_symbolic",
+    mutates_args=(),
+    schema="(Tensor[] inputs, str op_type, int onnx_dtype, Tensor[] attr_tensors, *, SymInt[] shape, str[] attr_keys, str[] attr_types, int[][] attr_pos, int[] attr_ints, float[] attr_floats, str[] attr_strs, str[] metadata_props_keys, str[] metadata_props_values, str domain='', int? version=None) -> Tensor",
+)
+def _symbolic(
+    inputs: Sequence[torch.Tensor],
+    op_type: str,
+    onnx_dtype: int,
+    attr_tensors: Sequence[torch.Tensor],
+    *,
+    shape: Sequence[Union[int, torch.SymInt]],
+    attr_keys: Sequence[str],
+    attr_types: Sequence[str],
+    attr_pos: Sequence[tuple[int, int]],
+    attr_ints: Sequence[int],
+    attr_floats: Sequence[float],
+    attr_strs: Sequence[str],
+    metadata_props_keys: Sequence[str] = (),
+    metadata_props_values: Sequence[str] = (),
+    domain: str = "",
+    version: Optional[int] = None,
+) -> torch.Tensor:
+    # TODO: Verify that shape supports SymInt
+    torch._check(
```

Comment:
Is this the correct way to use torch._check? Have we confirmed it's effect on constraints?

Reply:
Sorry I read your question wrong. I think I saw similar usages like https://github.com/pytorch/pytorch/blob/eb7bf4202dec7c9bdfa2d308d5e65b0dc93f89da/torch/_refs/fft.py#L74 . This helps propagating the error when torchdynamo is being used. Here we are not asserting any dynamic shapes so I hope we are ok.

Reply:
We are using it in registering fake level here. IS https://github.com/pytorch/pytorch/blob/eb7bf4202dec7c9bdfa2d308d5e65b0dc93f89da/torch/_refs/fft.py#L58 using it in registering fake level as well?